src/fhirtypes/dataTypes.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  InvalidRangeValueError,
+  UnitMismatchError,
+  CodeAndSystemMismatchError,
+  InvalidPeriodError
+} from '../errors';
+import { FHIRDateTime, validateFHIRDateTime } from './primitiveTypes';
+
+export type CodeableConcept = {
```

Comment:
I really like how you documented (in comments) the other types and included the links.  I (obviously) did not do that in the types I initially created, but could you add that documentation here so they're all documented consistently?